### PR TITLE
[RHPAM-2969] - Custom Database Image for Sybase not working

### PR DESCRIPTION
**Pull request by spolti**
Signed-off-by: spolti <fspolti@redhat.com>

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [x] Pull Request title is properly formatted: `[RHPAM-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Attached commits represent units of work and are properly formatted
- [x] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [x] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

**Comment by spolti** [comment]
@desmax74 @jakubschwan @Sgitario please review.

File: templates/contrib/jdbc/README.md [hunk 1]
```diff
@@ -68,17 +68,17 @@ Driver version: 42.2.5
...
 oc create -f extension-image-streams.yaml -m <NAME_SPACE>
 ```
 
 The namespace is the project that you are working on OpenShift, or, you can also choose a common namespace to install these imagestreams, like `openshift`
 
-## Building a extension image
+## Building an extension image
 
-All you need to do is install cekit2 and execute the `make build` command specifying the image you want to build, i.e.:
+All you need to do is install [CeKit](https://docs.cekit.io/en/latest/handbook/installation/instructions.html) and execute the `make build` command specifying the image you want to build, i.e.:
```
Comment (jakubschwan) [inline templates/contrib/jdbc/README.md:81]: I think there should be mentioned also cekit version. Kie Server is build with 3.2. With newer versions there is the issue with max depth reached for layers.

Comment (spolti) [reply templates/contrib/jdbc/README.md:81]: In this case I don't think it really matters, lastest version is good enough.

Comment (spolti) [reply templates/contrib/jdbc/README.md:81]: actually, there is a little inconpatibility, I fixed it as part of https://issues.redhat.com/browse/RHPAM-2930

**Comment by Sgitario** [comment]
I tested these changes and I'm now able to create the sybase image and this image is working fine with Kie Server. Thanks @spolti !

**Comment by spolti** [comment]
added a comment in the jira, seems not be related with the jdbc driver.

File: templates/contrib/jdbc/README.md (hunk 1 above)
Comment (spolti) [reply templates/contrib/jdbc/README.md:81]: @jakubschwan I updated it https://github.com/jboss-container-images/rhpam-7-openshift-image/pull/444
for this PR let's use cekit 3.2, later we can update it.

**Comment by Sgitario** [comment]
> After checking this change, the Kie Server is failing to connect to the Sybase database with this error:
> 
> ```
> �[0m�[31m10:39:52,131 ERROR [org.kie.server.services.impl.KieServerImpl] (ServerService Thread Pool -- 78) Error when initializing server extension of type jBPM KIE Server extension: java.lang.RuntimeException: Unable to create EntityManagerFactory due to [PersistenceUnit: org.jbpm.domain] Unable to build Hibernate SessionFactory
> 	at deployment.ROOT.war//org.kie.server.services.jbpm.JbpmKieServerExtension.build(JbpmKieServerExtension.java:803)
> 	at deployment.ROOT.war//org.kie.server.services.jbpm.JbpmKieServerExtension.configureServices(JbpmKieServerExtension.java:231)
> 	at deployment.ROOT.war//org.kie.server.services.jbpm.JbpmKieServerExtension.init(JbpmKieServerExtension.java:194)
> 	at deployment.ROOT.war//org.kie.server.services.impl.KieServerImpl.init(KieServerImpl.java:165)
> 	at deployment.ROOT.war//org.kie.server.services.impl.KieServerLocator$KieServerLocatorHelper.<clinit>(KieServerLocator.java:25)
> 	at deployment.ROOT.war//org.kie.server.services.impl.KieServerLocator.getInstance(KieServerLocator.java:32)
> 	at deployment.ROOT.war//org.kie.server.remote.rest.common.Bootstrap.contextInitialized(Bootstrap.java:51)
> 	at io.undertow.servlet@2.0.28.SP1-redhat-00001//io.undertow.servlet.core.ApplicationListeners.contextInitialized(ApplicationListeners.java:187)
> 	at io.undertow.servlet@2.0.28.SP1-redhat-00001//io.undertow.servlet.core.DeploymentManagerImpl$1.call(DeploymentManagerImpl.java:217)
> 	at io.undertow.servlet@2.0.28.SP1-redhat-00001//io.undertow.servlet.core.DeploymentManagerImpl$1.call(DeploymentManagerImpl.java:186)
> 	at io.undertow.servlet@2.0.28.SP1-redhat-00001//io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:42)
> 	at io.undertow.servlet@2.0.28.SP1-redhat-00001//io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
> 	at org.wildfly.extension.undertow@7.2.6.GA-redhat-00001//org.wildfly.extension.undertow.security.SecurityContextThreadSetupAction.lambda$create$0(SecurityContextThreadSetupAction.java:105)
> 	at org.wildfly.extension.undertow@7.2.6.GA-redhat-00001//org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1504)
> 	at org.wildfly.extension.undertow@7.2.6.GA-redhat-00001//org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1504)
> 	at org.wildfly.extension.undertow@7.2.6.GA-redhat-00001//org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1504)
> 	at org.wildfly.extension.undertow@7.2.6.GA-redhat-00001//org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1504)
> 	at io.undertow.servlet@2.0.28.SP1-redhat-00001//io.undertow.servlet.core.DeploymentManagerImpl.deploy(DeploymentManagerImpl.java:252)
> 	at org.wildfly.extension.undertow@7.2.6.GA-redhat-00001//org.wildfly.extension.undertow.deployment.UndertowDeploymentService.startContext(UndertowDeploymentService.java:96)
> 	at org.wildfly.extension.undertow@7.2.6.GA-redhat-00001//org.wildfly.extension.undertow.deployment.UndertowDeploymentService$1.run(UndertowDeploymentService.java:78)
> 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
> 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
> 	at org.jboss.threads@2.3.2.Final-redhat-1//org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
> 	at org.jboss.threads@2.3.2.Final-redhat-1//org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1985)
> 	at org.jboss.threads@2.3.2.Final-redhat-1//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1487)
> 	at org.jboss.threads@2.3.2.Final-redhat-1//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1378)
> 	at java.base/java.lang.Thread.run(Thread.java:834)
> 	at org.jboss.threads@2.3.2.Final-redhat-1//org.jboss.threads.JBossThread.run(JBossThread.java:485)
> Caused by: javax.persistence.PersistenceException: [PersistenceUnit: org.jbpm.domain] Unable to build Hibernate SessionFactory
> 	at deployment.ROOT.war//org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.persistenceException(EntityManagerFactoryBuilderImpl.java:1016)
> 	at deployment.ROOT.war//org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.build(EntityManagerFactoryBuilderImpl.java:942)
> 	at deployment.ROOT.war//org.hibernate.jpa.HibernatePersistenceProvider.createContainerEntityManagerFactory(HibernatePersistenceProvider.java:141)
> 	at deployment.ROOT.war//org.kie.server.services.jbpm.JbpmKieServerExtension.createEntityManagerFactory(JbpmKieServerExtension.java:808)
> 	at deployment.ROOT.war//org.kie.server.services.jbpm.JbpmKieServerExtension.build(JbpmKieServerExtension.java:801)
> 	... 27 more
> Caused by: org.hibernate.exception.GenericJDBCException: Unable to open JDBC Connection for DDL execution
> 	at deployment.ROOT.war//org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:47)
> 	at deployment.ROOT.war//org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:113)
> 	at deployment.ROOT.war//org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:99)
> 	at deployment.ROOT.war//org.hibernate.resource.transaction.backend.jta.internal.DdlTransactionIsolatorJtaImpl.<init>(DdlTransactionIsolatorJtaImpl.java:62)
> 	at deployment.ROOT.war//org.hibernate.resource.transaction.backend.jta.internal.JtaTransactionCoordinatorBuilderImpl.buildDdlTransactionIsolator(JtaTransactionCoordinatorBuilderImpl.java:46)
> 	at deployment.ROOT.war//org.hibernate.tool.schema.internal.HibernateSchemaManagementTool.getDdlTransactionIsolator(HibernateSchemaManagementTool.java:175)
> 	at deployment.ROOT.war//org.hibernate.tool.schema.internal.AbstractSchemaMigrator.doMigration(AbstractSchemaMigrator.java:94)
> 	at deployment.ROOT.war//org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.performDatabaseAction(SchemaManagementToolCoordinator.java:183)
> 	at deployment.ROOT.war//org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.process(SchemaManagementToolCoordinator.java:72)
> 	at deployment.ROOT.war//org.hibernate.internal.SessionFactoryImpl.<init>(SessionFactoryImpl.java:310)
> 	at deployment.ROOT.war//org.hibernate.boot.internal.SessionFactoryBuilderImpl.build(SessionFactoryBuilderImpl.java:467)
> 	at deployment.ROOT.war//org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.build(EntityManagerFactoryBuilderImpl.java:939)
> 	... 30 more
> Caused by: java.sql.SQLException: javax.resource.ResourceException: IJ000453: Unable to get managed connection for java:jboss/datasources/jbpmDS
> 	at org.jboss.ironjacamar.jdbcadapters@1.4.18.Final-redhat-00001//org.jboss.jca.adapters.jdbc.WrapperDataSource.getConnection(WrapperDataSource.java:159)
> 	at org.jboss.as.connector@7.2.6.GA-redhat-00001//org.jboss.as.connector.subsystems.datasources.WildFlyDataSource.getConnection(WildFlyDataSource.java:64)
> 	at deployment.ROOT.war//org.hibernate.engine.jdbc.connections.internal.DatasourceConnectionProviderImpl.getConnection(DatasourceConnectionProviderImpl.java:122)
> 	at deployment.ROOT.war//org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentInitiator$ConnectionProviderJdbcConnectionAccess.obtainConnection(JdbcEnvironmentInitiator.java:180)
> 	at deployment.ROOT.war//org.hibernate.resource.transaction.backend.jta.internal.DdlTransactionIsolatorJtaImpl.<init>(DdlTransactionIsolatorJtaImpl.java:59)
> 	... 38 more
> Caused by: javax.resource.ResourceException: IJ000453: Unable to get managed connection for java:jboss/datasources/jbpmDS
> 	at org.jboss.ironjacamar.impl@1.4.18.Final-redhat-00001//org.jboss.jca.core.connectionmanager.AbstractConnectionManager.getManagedConnection(AbstractConnectionManager.java:690)
> 	at org.jboss.ironjacamar.impl@1.4.18.Final-redhat-00001//org.jboss.jca.core.connectionmanager.tx.TxConnectionManagerImpl.getManagedConnection(TxConnectionManagerImpl.java:440)
> 	at org.jboss.ironjacamar.impl@1.4.18.Final-redhat-00001//org.jboss.jca.core.connectionmanager.AbstractConnectionManager.allocateConnection(AbstractConnectionManager.java:789)
> 	at org.jboss.ironjacamar.jdbcadapters@1.4.18.Final-redhat-00001//org.jboss.jca.adapters.jdbc.WrapperDataSource.getConnection(WrapperDataSource.java:151)
> 	... 42 more
> Caused by: javax.resource.ResourceException: IJ031084: Unable to create connection
> 	at org.jboss.ironjacamar.jdbcadapters@1.4.18.Final-redhat-00001//org.jboss.jca.adapters.jdbc.local.LocalManagedConnectionFactory.createLocalManagedConnection(LocalManagedConnectionFactory.java:345)
> 	at org.jboss.ironjacamar.jdbcadapters@1.4.18.Final-redhat-00001//org.jboss.jca.adapters.jdbc.local.LocalManagedConnectionFactory.getLocalManagedConnection(LocalManagedConnectionFactory.java:352)
> 	at org.jboss.ironjacamar.jdbcadapters@1.4.18.Final-redhat-00001//org.jboss.jca.adapters.jdbc.local.LocalManagedConnectionFactory.createManagedConnection(LocalManagedConnectionFactory.java:287)
> 	at org.jboss.ironjacamar.impl@1.4.18.Final-redhat-00001//org.jboss.jca.core.connectionmanager.pool.mcp.SemaphoreConcurrentLinkedDequeManagedConnectionPool.createConnectionEventListener(SemaphoreConcurrentLinkedDequeManagedConnectionPool.java:1328)
> 	at org.jboss.ironjacamar.impl@1.4.18.Final-redhat-00001//org.jboss.jca.core.connectionmanager.pool.mcp.SemaphoreConcurrentLinkedDequeManagedConnectionPool.getConnection(SemaphoreConcurrentLinkedDequeManagedConnectionPool.java:499)
> 	at org.jboss.ironjacamar.impl@1.4.18.Final-redhat-00001//org.jboss.jca.core.connectionmanager.pool.AbstractPool.getSimpleConnection(AbstractPool.java:632)
> 	at org.jboss.ironjacamar.impl@1.4.18.Final-redhat-00001//org.jboss.jca.core.connectionmanager.pool.AbstractPool.getConnection(AbstractPool.java:604)
> 	at org.jboss.ironjacamar.impl@1.4.18.Final-redhat-00001//org.jboss.jca.core.connectionmanager.AbstractConnectionManager.getManagedConnection(AbstractConnectionManager.java:624)
> 	... 45 more
> Caused by: java.sql.SQLException: JZ0D5: Error loading protocol com.sybase.jdbc4.//sybase-160.hosts.prod.upshift.rdu2.redhat.com.//sybase-160.hosts.prod.upshift.rdu2.redhat.com.
> 	at com.sybase//com.sybase.jdbc4.jdbc.ErrorMessage.raiseError(ErrorMessage.java:752)
> 	at com.sybase//com.sybase.jdbc4.jdbc.ProtocolManager.getProtocol(ProtocolManager.java:123)
> 	at com.sybase//com.sybase.jdbc4.jdbc.SybUrlManager.loadProtocol(SybUrlManager.java:324)
> 	at com.sybase//com.sybase.jdbc4.jdbc.SybUrlManager.init(SybUrlManager.java:222)
> 	at com.sybase//com.sybase.jdbc4.jdbc.SybUrlManager.getUrlProvider(SybUrlManager.java:164)
> 	at com.sybase//com.sybase.jdbc4.jdbc.SybDriver.connect(SybDriver.java:224)
> 	at org.jboss.ironjacamar.jdbcadapters@1.4.18.Final-redhat-00001//org.jboss.jca.adapters.jdbc.local.LocalManagedConnectionFactory.createLocalManagedConnection(LocalManagedConnectionFactory.java:321)
> 	... 52 more
> ```
> 
> I'm not sure if this issue is about this PR or something new, so I will update the ticket accordingly.

This issue was a bug in the test framework. I tested the PR changes after fixing them and everything went fine.

**Comment by spolti** [comment]
@Sgitario cool, thanks :)

**Pull request merged**